Automattic/wp-calypso
# Plans section: Modernized header and navbar / part 1

### Proposed Changes

Partially addresses https://github.com/Automattic/martech/issues/1463

This brings the "modernized" header & navbar by default to `/plans/my-plan` & `/plan` (not behind a feature flag). This is another temp solution with follow-ups in TODO list below.

_Notes:_ Plans header & navigation are used for [jetpack-plans](https://github.com/Automattic/wp-calypso/tree/trunk/client/my-sites/plans/jetpack-plans) code, but the changes should not affect these (something for a follow-up)

#### Media

with some of the states I notice for the plans pages (info & upsell notices):

_Old `/plans`_

<img width="700" alt="Screenshot 2023-02-06 at 4 25 02 PM" src="https://user-images.githubusercontent.com/1705499/216997389-4ab1c688-857d-4958-81a4-cbd73af1b934.png">

_New `/plans`_

<img width="700" alt="Screenshot 2023-02-06 at 4 23 42 PM" src="https://user-images.githubusercontent.com/1705499/216997457-c062689e-2417-4522-9ced-c2bfcef61345.png">

_New `/plans` free site_

<img width="700" alt="Screenshot 2023-02-06 at 5 12 03 PM" src="https://user-images.githubusercontent.com/1705499/217009326-c5995696-26ae-4186-a241-3fad2e70fc5f.png">


_`/plans/my-plan`_

<img width="700" alt="Screenshot 2023-02-06 at 4 24 30 PM" src="https://user-images.githubusercontent.com/1705499/216997597-9fc7b1b1-f348-4e16-94e5-b8010dc39895.png">

### Testing Instructions



- Navigate to `/plans/[site_slug]?flags=onboarding/2023-pricing-grid`
- Confirm the changes
- Do the same without the flag `/plans/[site_slug]?flags=-onboarding/2023-pricing-grid`
- Confirm no changes in `jetpack-plans` (and please post instructions in a comment on how to test that while at it 😄)

### TODO

- [x] Extract into a layout component to allow reuse | [issue](https://github.com/Automattic/wp-calypso/issues/73102)
- [x] Extract navigation into a component to allow reuse | [issue](https://github.com/Automattic/wp-calypso/issues/73102)
- [x] Consider navbar inside header to reduce some of the complexity | [issue](https://github.com/Automattic/wp-calypso/issues/73102)

### Pre-merge Checklist





Related to Automattic/martech#1463
